install.sh

```diff
@@ -1,90 +1,349 @@
+    if [[ " ${AVAILABLE_REGIONS[*]} " =~ " $AWS_DEFAULT_REGION " ]]; then
+      echo "Region $AWS_DEFAULT_REGION is enabled for your account."
+      break
+    else
+      echo "Error: Region $AWS_DEFAULT_REGION is not enabled for your account or invalid region code."
+    fi
+  fi
+done
+
 export LOG_FILE="cdk.services.log"
 function echoLog() {
   echo "$1" | tee -a $LOG_FILE
 }
-function isValidPass() {
-  if [[ ! $1 =~ ^([A-Z]|[a-z])([A-Z]|[a-z]|[0-9]){7,}$ ]]; then
-    echo "Error: Input does not match the pattern [A-Z][a-z][0-9] and should have at least 8 characters and start with alphabet."
-    exit 1
-  fi
+
+echo
+echo "${blue}##########################################${reset}"
+echo "${blue}       Installing Dependencies${reset}"
+echo "${blue}##########################################${reset}"
+echo
+# Function to display a simple loading animation
+show_loader() {
+    local message=$1
+    local pid=$!
+    local delay=0.3
+    local spinstr='|/-\\'
+    while [ "$(ps a | awk '{print $1}' | grep $pid)" ]; do
+        local temp=${spinstr#?}
+        printf "\r%s [%c]  " "$message" "$spinstr"
+        local spinstr=$temp${spinstr%"$temp"}
+        sleep $delay
+    done
+    printf "\r%s [Done]   \n" "$message"
 }
 
-echo "##########################################\nInstalling dependencies\n##########################################"
-# Install dependencies
-if ! command -v node &> /dev/null
-then
-    echo "node could not be found. Please install Node.js 18 or above."
+# Check for Node.js
+echo "Checking for Node.js..."
+if ! command -v node &> /dev/null; then
+    echo "Node.js could not be found. Please install node js 18 or above."
     exit 1
 fi
+
+# Verify Node.js version
 version=$(node -v | cut -d'.' -f1 | tr -d 'v')
-if [ "$version" -lt 18 ]
-then
-  echo "Invalid Node.js version. Expected 18 or above, but got $version."
-  exit 1
+if [ "$version" -lt 18 ]; then
+    echo "Invalid Node.js version. Expected 18 or above, but got $version."
+    exit 1
 fi
+echo "Node.js version is valid."
+
+# Check and Install AWS CLI
+echo "Checking for AWS CLI..."
 if aws --version; then
-  echo "##########################################\nAWS CLI Installed\n##########################################"
+    echo "AWS CLI is already installed."
 else
-  curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
-  sudo installer -pkg AWSCLIV2.pkg -target /
+    echo "Installing AWS CLI..."
+    curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
+    sudo installer -pkg AWSCLIV2.pkg -target / & show_loader "Installing AWS CLI.."
+    echo "AWS CLI is installed."
 fi
-if ! command -v aws &> /dev/null
-then
-    echo "AWS CLI could not be found. Please rerun \`sh install.sh\` with Sudo access"
+
+# Check if AWS CLI installation was successful
+if ! command -v aws &> /dev/null; then
+    echo "AWS CLI could not be found. Please rerun 'sh install.sh' with Sudo access."
```

Comment:
Wherever we are checking for the presence of a command using `command -v`, we might have to also suggest the user to ensure that the command is available within `$PATH`.